src/java/org/apache/cassandra/index/sai/disk/RAMPostingSlices.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk;
+
+import java.io.IOException;
+
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.ByteBlockPool;
+import org.apache.lucene.util.Counter;
+import org.apache.lucene.util.mutable.MutableValueInt;
+
+/**
+ * Encodes postings as variable integers into slices
+ */
+class RAMPostingSlices
+{
+    static final int DEFAULT_TERM_DICT_SIZE = 1024;
+
+    private final ByteBlockPool postingsPool;
+    private int[] postingStarts = new int[DEFAULT_TERM_DICT_SIZE];
+    private int[] postingUptos = new int[DEFAULT_TERM_DICT_SIZE];
+    private int[] sizes = new int[DEFAULT_TERM_DICT_SIZE];
```

Comment:
For both `RAMPostingSlices` and `ByteSliceReader`, I'd do two things:

1.) Add brief JavaDoc style comments for the important instance fields. (Here, `postingStarts` et al.)
2.) Once we have those, make sure we like the field names. (ex. `sizes` here could be something like `termPostingCounts`)

Perhaps an alternative to #1 would just be describing the relationship between starts, up-tos, and sizes in a class-level JavaDoc.